charset/charset.go

```diff
@@ -69,17 +67,14 @@ var supportedCollationNames = map[string]struct{}{
 	CollationBin:     {},
 }
 
-// Desc is a charset description.
-type Desc struct {
-	Name             string
-	Desc             string
-	DefaultCollation string
-	Maxlen           int
-}
-
 // GetSupportedCharsets gets descriptions for all charsets supported so far.
-func GetSupportedCharsets() []*Desc {
-	return descs
+func GetSupportedCharsets() []*Charset {
```

Comment:
This change brings more memory allocations, can we avoid them?

Reply:
actually, I think it's fine, I search the usage of this function in tidb, it is only used for SQL like `show charset`.
I believe there are more codes like this, it is not a big problem. If we want to avoid it anyway, we must maintain another slice to hold the same information, it hurts the brain.

Reply:
> I search the usage of this function in tidb, it is only used for SQL like show charset.

Ok.

> it hurts the brain.

I don't think this is a problem. We can encapsulate the slice into a struct to keep the readability.